### ISSUE TITLE
Terminal does not redraw after application inside it scrolls

### ISSUE THREAD

**Issue by christopherdumas** [SerenityOS/serenity#136]
See below

**Comment by christopherdumas** [comment]
Every time this happens, the debug log says:
```
FIXME: escape$t: Ps: 22
FIXME: escape$r: Set scrolling region: 1-24
```
but I don't know what `escape$t` is meant to be. I'll fix `escape$r` first.

**Comment by christopherdumas** [comment]
I've fixed this. It's related to Terminal's incomplete terminal emulation capabilities. A PR fixing that is upcoming.

**Comment by christopherdumas** [comment]
Okay! So it appears that although the actual contents of the terminal's lines are in fact updated correctly by Vim, only the first (or last) line of the screen is ever being redrawn. This leads to hilarious issues with scrolling in Vim, which was the main compatibility issue. I assumed this was a function of not implementing enough TTY stuff, but that was not the case. This is just a bug of incomplete implementation in what we already have. So, when I scroll down from here:

![2019-05-28-135042_1080x796_scrot](https://user-images.githubusercontent.com/1920151/58511533-d9eb8080-814f-11e9-852f-3ad16c9a2d43.png)

To here:
![2019-05-28-135051_1080x796_scrot](https://user-images.githubusercontent.com/1920151/58511574-f5ef2200-814f-11e9-8e61-23c560732e7b.png)

The bottom line is simply... drawn over. `force_repaint` and `update` do not work. Meaning, when I call them, they just repaint the top/bottom line. They do *nothing*. However, if I do <kbd>Ctrl</kbd>+<kbd>L</kbd> in the terminal, everything redraws and looks correct.

There are two options that I can see here:
1. The internal lines are actually not correct, and when it scrolls its actually blanking out everything but the first/last line, and that's why its not drawing, and <kbd>Ctrl</kbd>+<kbd>L</kbd> does something totally different that I wasn't aware of
2. Redrawing is broken.

I'm not really sure which it is.

**Comment by christopherdumas** [comment]
Okay so the lines aren't being blanked, they're there.

**Comment by awesomekling** [comment]
This doesn't appear to be an issue anymore :)
